### PR TITLE
Adds automated ACME tests using Caddy.

### PR DESCRIPTION
**Pull request by schultz-is**
# Summary
Tests Vault issuing certificates to Caddy via ACME. Includes coverage for HTTP challenges, EAB functionality, and TLS-ALPN challenges.

### CODE REVIEW

File: builtin/logical/pkiext/pkiext_binary/acme_test.go [hunk 1]
```diff
@@ -65,6 +81,161 @@ func Test_ACME(t *testing.T) {
...
+
+		ipAddr := networks[vaultNetwork]
+		hostname := fmt.Sprintf("%s.dadgarcorp.com", runID)
+
+		err = pki.AddHostname(hostname, ipAddr)
+		require.NoError(t, err, "failed to update vault host files")
+
+		// Copy the Vault cluster's CA certificate over to the Caddy container.
+		caCertCtx := hDocker.NewBuildContext()
+		caCertCtx["vault_ca_cert.crt"] = hDocker.PathContentsFromString(string(cluster.GetCACertPEM()))
```
Comment (cipherboy) [inline builtin/logical/pkiext/pkiext_binary/acme_test.go:161]: This and the push below could be combined fwiw I believe.

File: builtin/logical/pkiext/pkiext_binary/pki_cluster.go [hunk 2]
```diff
@@ -109,6 +109,11 @@ func (vpc *VaultPkiCluster) GetActiveNode() *api.Client {
 	return vpc.GetActiveClusterNode().APIClient()
 }
 
+// GetCACertPEM returns the Vault cluster's PEM-encoded CA certificate.
+func (vpc *VaultPkiCluster) GetCACertPEM() []byte {
```
Comment (cipherboy) [inline builtin/logical/pkiext/pkiext_binary/pki_cluster.go:113]: Maybe `GetListenerCACertPEM()` to disambiguate it?

**Review by cipherboy — APPROVED** [review]

**Review by stevendpclark — APPROVED** [review]

File: builtin/logical/pkiext/pkiext_binary/acme_test.go (hunk 1 above)
Comment (schultz-is) [reply builtin/logical/pkiext/pkiext_binary/acme_test.go:161]: Oh, good catch! I'll combine them

File: builtin/logical/pkiext/pkiext_binary/pki_cluster.go (hunk 2 above)
Comment (schultz-is) [reply builtin/logical/pkiext/pkiext_binary/pki_cluster.go:113]: I like that suggestion. I was trying to come up with a distinguishing name, but was drawing blanks.

**Review by stevendpclark — APPROVED** [review]

**Review by cipherboy — APPROVED** [review]
:rocket:

**Pull request merged**